Add tests for already merged classes of RDE library - PR3

isorrentino opened the pull request:
This PR implements the UkfState and UkfMeasurment classes which represent the UKF process and measurement models.
It also adds the final RDE library that can be called from any application.

This branch is rebased on https://github.com/isorrentino/bipedal-locomotion-framework/tree/RDE/PR2. I convert the PR to draft until https://github.com/ami-iit/bipedal-locomotion-framework/pull/742 is merged.

GiulioRomualdi commented:
I will wait for a PR with the tests that have been removed in https://github.com/ami-iit/bipedal-locomotion-framework/pull/742 before reviewing this PR

isorrentino commented:
Hi @GiulioRomualdi, I have updated the title of this PR and changed the status to `Ready for Review`. Additionally, I have added tests for the classes that have already been merged into the RDE library. Furthermore, I have removed the classes without tests. They will be re-added in a new pull request along with their corresponding tests.

GiulioRomualdi commented on src/Estimators/tests/RobotDynamicsEstimator/CMakeLists.txt:
```diff
@@ -29,6 +36,54 @@ if(FRAMEWORK_USE_icub-models AND FRAMEWORK_COMPILE_YarpImplementation)
+        Eigen3::Eigen
+        MANIF::manif
+        BipedalLocomotion::ManifConversions
+        ${iDynTree_LIBRARIES})
```
Can you specify the libraries of iDynTree that you use?

GiulioRomualdi commented on src/Estimators/tests/RobotDynamicsEstimator/FrictionTorqueStateDynamicsTest.cpp:
```diff
@@ -0,0 +1,328 @@
+#include <BipedalLocomotion/Math/Constants.h>
+#include <BipedalLocomotion/ParametersHandler/IParametersHandler.h>
+#include <BipedalLocomotion/ParametersHandler/StdImplementation.h>
+#include <BipedalLocomotion/ParametersHandler/YarpImplementation.h>
```
Looking at the code it seems that you do not use it. I think you can remove it from here and also from the cmake (including the part of the if statement in the CMakeLists.txt)

GiulioRomualdi replied:
Nope you have to keep the if statement for some of the tests but the tests that do not depend on it you can move them away
